src/helix/endpoints/chat/send_chat_message.rs

```diff
@@ -0,0 +1,266 @@
+    pub broadcaster_id: Cow<'a, types::UserIdRef>,
+    /// The ID of the user sending the message. This ID must match the user ID in the user access token.
+    #[cfg_attr(feature = "typed-builder", builder(setter(into)))]
+    #[cfg_attr(feature = "deser_borrow", serde(borrow = "'a"))]
+    pub sender_id: Cow<'a, types::UserIdRef>,
+    /// The message to send.
+    ///
+    /// The message is limited to a maximum of 500 characters.
+    /// Chat messages can also include emoticons.
+    /// To include emoticons, use the name of the emote. The names are case sensitive.
+    /// Don't include colons around the name (e.g., :bleedPurple:).
+    /// If Twitch recognizes the name, Twitch converts the name to the emote
+    /// before writing the chat message to the chat room.
+    #[cfg_attr(feature = "typed-builder", builder(setter(into)))]
+    #[cfg_attr(feature = "deser_borrow", serde(borrow = "'a"))]
+    pub message: Cow<'a, str>,
+    /// The ID of the chat message being replied to.
+    #[cfg_attr(feature = "typed-builder", builder(setter(into)))]
+    #[cfg_attr(feature = "deser_borrow", serde(borrow = "'a"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub reply_parent_message_id: Option<Cow<'a, types::MsgIdRef>>,
+}
+
+impl<'a> SendChatMessageBody<'a> {
+    /// Send a message in the broadcaster's channel
+    pub fn new(
+        broadcaster_id: impl types::IntoCow<'a, types::UserIdRef> + 'a,
+        sender_id: impl types::IntoCow<'a, types::UserIdRef> + 'a,
+        message: impl types::IntoCow<'a, str> + 'a,
+    ) -> Self {
+        Self {
+            broadcaster_id: broadcaster_id.into_cow(),
+            sender_id: sender_id.into_cow(),
+            message: message.into_cow(),
+            reply_parent_message_id: None,
+        }
+    }
+
+    /// Set the reply parent message-id
+    pub fn reply_parent_message_id(
+        mut self,
+        reply_parent_message_id: impl types::IntoCow<'a, types::MsgIdRef> + 'a,
+    ) -> Self {
+        self.reply_parent_message_id = Some(reply_parent_message_id.into_cow());
+        self
+    }
+}
+
+impl helix::private::SealedSerialize for SendChatMessageBody<'_> {}
+
+impl<'a> helix::HelixRequestBody for [SendChatMessageBody<'a>] {
+    fn try_to_body(&self) -> Result<hyper::body::Bytes, helix::BodyError> {
+        #[derive(Serialize)]
+        struct InnerBody<'a> {
+            data: &'a [SendChatMessageBody<'a>],
+        }
+
+        serde_json::to_vec(&InnerBody { data: self })
+            .map_err(Into::into)
+            .map(Into::into)
+    }
+}
+
+/// A drop reason of a sent message.
+#[derive(PartialEq, Eq, Deserialize, Serialize, Debug, Clone)]
+#[cfg_attr(feature = "deny_unknown_fields", serde(deny_unknown_fields))]
+#[non_exhaustive]
+pub struct ChatMessageDropReason {
+    /// Code for why the message was dropped.
+    pub code: String,
+    /// Message for why the message was dropped.
+    pub message: String,
+}
```

Comment:
This could probably use examples of how what the `code` can look like, `msg_rejected` etc. Maybe even make `code` an enum with a catch-all